shared/src/main/scala/mlscript/ConstraintSolver.scala

```diff
@@ -34,8 +34,9 @@ class ConstraintSolver extends NormalForms { self: Typer =>
     val info = ctx.tyDefs2.getOrElse(clsNme, die/*TODO*/)
     
     if (info.isComputing) {
-      
-      ??? // TODO support?
+
+      L(ErrorReport(
+        msg"${info.decl.kind.str.capitalize} `${info.decl.name}` is not supported yet." -> fld.toLoc :: Nil))
```

Comment:
How could this be the appropriate error here? Do you have an example?

Reply:
I now see that you meant to fix this example:

```ts
module M {
  class A
  fun a: M.A
  fun a = 1
}
//│ /!!!\ Uncaught error: scala.NotImplementedError: an implementation is missing
//│ ╔══[ERROR] Module `M` is not supported yet.
//│ ║  l.170: 	  fun a: M.A
//│ ╙──       	          ^^
```

That's clearly not the correct error. For now you could make it:
```suggestion
      L(ErrorReport(
        // TODO improve
        msg"${info.decl.kind.str.capitalize} `${info.decl.name}` has a cyclic type dependency that is not supported yet." -> fld.toLoc :: Nil))
```